hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/VersionedInCurrentVersionTest.java

```diff
@@ -148,4 +147,17 @@ public void testNewVersionedClass_doesNotInvokeLessThan_whenReadingData() {
 
         return previousVersionReflections.getSubTypesOf(Versioned.class);
     }
+
+    private File[] getPreviousVersionArtifacts() throws Exception {
+        File previousVersionFolder = temporaryFolder.newFolder();
+        try {
+            return HazelcastVersionLocator.locateVersion(
+                    PREVIOUS_CLUSTER_VERSION.toString(),
+                    previousVersionFolder,
+                    getBuildInfo().isEnterprise());
+        } catch (GuardianException e) {
+            assumeNoException("The requested version could not be downloaded, most probably it has not been released yet", e);
```

Comment:
This is the actual fix for the test. It skips the test when an exception is thrown here.

At the moment the test tries to download the released HZ 3.10 version to compare the operations with 3.11-SNAPSHOT. This won't work until the 3.10 version is released. This will throw a `GuardianException` exception, so the `assumeNoException()` will skip the actual test, since the exception is not `null`.

I tried to add the snapshot repository to the `HazelcastVersionLocator`, but we cannot guess the correct filename (there is no `hazelcast-3.10-SNAPSHOT.jar` in the repo, all JAR files have a timestamp). I don't think we can fix this test until we use something like Maven Invoker or Maven Embedder. So I guess this is the best option for now (and all upcoming windows, where we just created a new SNAPSHOT version and wait for the release of the previous one).